persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/EditTitleServiceTest.java

```diff
@@ -0,0 +1,182 @@
+import org.jdom2.output.XMLOutputter;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+
+import java.io.*;
+import java.util.Collection;
+import java.util.UUID;
+
+import static edu.unc.lib.dl.xml.JDOMNamespaceUtil.MODS_V3_NS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class EditTitleServiceTest {
+
+    @Mock
+    private AccessControlService aclService;
+    @Mock
+    private AgentPrincipals agent;
+    @Mock
+    private AccessGroupSet groups;
+    @Mock
+    private RepositoryObjectLoader repoObjLoader;
+    @Mock
+    private ContentObject contentObj;
+    @Mock
+    private BinaryObject binaryObj;
+    @Mock
+    private OperationsMessageSender operationsMessageSender;
+    @Mock
+    private MODSValidator modsValidator;
+
+    @Captor
+    private ArgumentCaptor<Collection<PID>> pidsCaptor;
+
+    private EditTitleService service;
+    private PID pid;
+    private Document document;
+
+    @Before
+    public void init() throws Exception {
+        initMocks(this);
+
+        pid = PIDs.get(UUID.randomUUID().toString());
+        service = new EditTitleService();
+
+        service.setAclService(aclService);
+        service.setModsValidator(modsValidator);
+        service.setRepoObjLoader(repoObjLoader);
+        service.setOperationsMessageSender(operationsMessageSender);
+
+        when(repoObjLoader.getRepositoryObject(eq(pid))).thenReturn(contentObj);
+        when(contentObj.getDescription()).thenReturn(binaryObj);
+        when(aclService.hasAccess(any(PID.class), any(AccessGroupSet.class), eq(Permission.editDescription)))
+                .thenReturn(true);
+        when(repoObjLoader.getRepositoryObject(any(PID.class))).thenReturn(contentObj);
+        when(agent.getPrincipals()).thenReturn(groups);
+        when(agent.getUsername()).thenReturn("agentname");
+        when(operationsMessageSender.sendUpdateDescriptionOperation(anyString(), any(Collection.class))).thenReturn("message_id");
+
+        document = new Document();
+    }
+
+    @Test
+    public void editTitleTest() throws Exception {
+        String title = "new title";
+        document.addContent(new Element("mods", MODS_V3_NS)
+                .addContent(new Element("titleInfo", MODS_V3_NS)
+                        .addContent(new Element("title", MODS_V3_NS).setText("original title"))));
+        when(binaryObj.getBinaryStream()).thenReturn(convertDocumentToStream(document));
+
+        service.editTitle(agent, pid, title);
+
+        verify(operationsMessageSender).sendUpdateDescriptionOperation(anyString(), pidsCaptor.capture());
+        Collection<PID> pids = pidsCaptor.getValue();
+        assertEquals(pids.size(), 1);
+        assertTrue(pids.contains(pid));
```

Comment:
you can `verify` that `obj.setDescription` was called, using an argument captor to get back the inputstream, and then deserialize it into a Document to verify that the title has changed. You'll need to do the same thing in many tests, so I'd suggest making a helper methods `getUpdatedDescriptionDocument` which does the argument captor stuff and deserializes it, and another named `assertTitleValue(document, expectedTitle)` to assert that there is a title with the expected value